src/components/textInput/TextInput.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { View, TextInput } from 'react-native'
+
+import styles from './style'
+import { AppColors } from '../../constants/constants'
+
+class CustomTextInput extends React.Component {
+    constructor(props) {
+        super(props)
+    }
+
+    render() {
+        const {
+            Icon,
+            value,
+            onChangeText,
+            placeholderText,
+            iconRight = false,
+            secureTextEntry = false,
+            style = styles.background,
+            textStyle = styles.textInput,
+        } = this.props
+        return (
+            <View style={style}>
+                {!iconRight ? Icon : null}
```

Comment:
It is the wrong way to use the ternary operator.
You can check is Icon exists and use it.